parser/helper/helpers.go

```diff
@@ -172,3 +173,46 @@ func (h *Helper) GetMethodName(msg *parser.LotusMessage, height int64, key filTy
 func (h *Helper) GetEVMSelectorSig(ctx context.Context, selectorID string) (string, error) {
 	return h.actorCache.GetEVMSelectorSig(ctx, selectorID)
 }
+
+func (h *Helper) FilterTxsByActorType(ctx context.Context, txs []*types.Transaction, actorType string) ([]*types.Transaction, error) {
+	var result []*types.Transaction
+	for _, tx := range txs {
+		addrTo, err := address.NewFromString(tx.TxTo)
+		if err != nil {
+			h.logger.Sugar().Errorf("could not parse address. Err: %s", err)
+			return nil, err
+		}
+		addrFrom, err := address.NewFromString(tx.TxFrom)
+		if err != nil {
+			h.logger.Sugar().Errorf("could not parse address. Err: %s", err)
+			return nil, err
+		}
+
+		// TODO: TipsetKey
+		isType, err := h.isAnyAddressOfType(ctx, []address.Address{addrTo, addrFrom}, int64(tx.Height), filTypes.EmptyTSK, actorType)
+		if err != nil {
+			h.logger.Sugar().Errorf("could not get actor type from address. Err: %s", err)
+			continue
+		}
+		if !isType {
+			continue
+		}
+
+		result = append(result, tx)
+	}
+
+	return result, nil
+}
+
+func (h *Helper) isAnyAddressOfType(_ context.Context, addresses []address.Address, height int64, key filTypes.TipSetKey, actorType string) (bool, error) {
+	for _, addr := range addresses {
+		actorName, err := h.GetActorNameFromAddress(addr, height, key)
```

Comment:
Is this caching the response internally, right? Just to be sure.

Reply:
yep